lib/resty/http.lua

```diff
@@ -725,7 +726,7 @@ function _M.send_request(self, params)
             headers["Host"] = self.host
         end
     end
-    if not headers["User-Agent"] then
+    if use_default_user_agent ~= false and not headers["User-Agent"] then
         headers["User-Agent"] = _M._USER_AGENT
```

Comment:
`use_default_user_agent ` maybe `nil`

Reply:
> `use_default_user_agent ` maybe `nil`

I think this code can be triggered when use_default_user_agent is `nil` or `true`